ni_measurementlink_service/_featuretoggles.py

```diff
@@ -141,3 +141,11 @@ def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> _T:
         return wrapper
 
     return decorator
+
```

Comment:
Is this ready to be enabled by a decouple .env file? I'm just checking my understanding of where we're at on that front. And if so, would you put "MEASUREMENTLINK_SESSION_MANAGEMENT_2024Q1_ENABLE" in the file? Where would this file be placed?

Reply:
Speaking to how python-decouple works:

> And if so, would you put "MEASUREMENTLINK_SESSION_MANAGEMENT_2024Q1_ENABLE" in the file?

You would put:
```
SESSION_MANAGEMENT_2024Q1=True
```
(or "true", or "T", "1" -> these are all cast to boolean True, while basically any other value is cast to False)

Anywhere from your cwd up to the next to the .git folder at the top of your Git repo (or, if you're not in a Git repo, up to the top of your current drive).

# Also

`python-decouple` evaluation process:
find and load a `.env` or `setttings.ini` file (IIRC, first found wins, but I don't remember order of preference between the file names). Examine the environment variables, and overwrite anything from the file with those values.

When asked (via the function call), perform a string fetch (or cast/convert to the desired type)

Reply:
This line appears to be looking for MEASUREMENTLINK_ENABLE_<feature toggle name>:
https://github.com/ni/measurementlink-python/pull/402/files#diff-bf2ce48b68e8ea6ae8c565e0145a4198b87e3574c327848571ae34e7fd023f40R102

Reply:
Yes, set `MEASUREMENTLINK_ENABLE_SESSION_MANAGEMENT_2024Q1=1`. Or `=True`. It accepts both.

I am going to create an `.env.sample` file for configuration settings, but I'm not sure I want to put all of the pre-release feature toggles in it.

Reply:
oh, because we did a wrapper...

ok, so..., `_PREFIX` is `MEASUREMENTLINK`, and the string passed in is the `name`, so it would be:
`if config(f"{_PREFIX}_ENABLE_{name}", default=False, cast=bool):`
turns into
`if config(f"MEASUREMENTLINK_ENABLE_SESSION_MANAGEMENT_2024Q1", default=False, cast=bool):`

so then the .env file or environment vars would look like:
```
MEASUREMENTLINK_ENABLE_SESSION_MANAGEMENT_2024Q1=True
```

Reply:
> while basically any other value is cast to False

It only accepts certain false values.
https://github.com/HBNetwork/python-decouple/blob/master/decouple.py#L29

Reply:
> It only accepts certain false values.
> https://github.com/HBNetwork/python-decouple/blob/master/decouple.py#L29

Thank you for the correction, I've updated the original note.

Reply:
Ok, I added `.env.sample` and tested it (which revealed that the default search path isn't what we want)

Reply:
Looks good! I like it. On the naming of the file, it just has to start with .env? Just want to be clear what's allowed. E.g. does '.environment.mysettings' work? Or settings.env?

Reply:
Just `.env` or `settings.ini` -> https://github.com/HBNetwork/python-decouple/blob/7c9b250c53bdd0ae96f449d5d2be918ed90ba273/decouple.py#L202C1-L205C7

https://pypi.org/project/python-decouple/#usage

Reply:
That code seems to mean ".env.sample" won't work? So should we add some instruction in the .env.sample to say "Rename this file to .env"?

Reply:
or in the readme? -> but yeah, I would expect so.

Reply:
I updated the instructions in `.env.sample` and will update the wiki in a bit.

I'm going to hold off on updating README.md until we update the examples.

Reply:
Slight correction to my earlier comment, `decouple.config` apparently loads the file lazily: 
> [Since config is lazy and only opens the configuration file when it’s first needed, you have the chance to change its encoding right after import.](https://pypi.org/project/python-decouple/#:~:text=Since%20config%20is%20lazy%20and%20only%20opens%20the%20configuration%20file%20when%20it%E2%80%99s%20first%20needed%2C%20you%20have%20the%20chance%20to%20change%20its%20encoding%20right%20after%20import.)